### ISSUE TITLE
Unsupported screen size  [1284, 2778] for path ....

### ISSUE THREAD

**Issue by fguchelaar** [fastlane/fastlane#17583]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
It appears that  changed it's screenshot specifications for the 6.5" display; resulting in an error when trying to upload screenshots with deliver. It's now expected to provide 1284 x 2778 (2778 x 1284) instead of 1242 x 2688 (2688 x 1242).

For testing, I simply changed the local values in my `/deliver/lib/deliver/app_screenshot.rb` and that seemed to do the trick. Should I conjure up a PR, or are these values used in other spaces as well, that need further investigation?

Also, the 6.1" display seems to be removed from App Store Connect.

##### Command executed

```
fastlane deliver --skip_metadata --skip_binary_upload --skip_app_version_update --overwrite_screenshots --force
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>% fastlane deliver --skip_metadata --skip_binary_upload --skip_app_version_update --overwrite_screenshots --force --verbose
[✔] 🚀 
[16:52:57]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
Successfully loaded Appfile at path '/Users/***/Workspace/github/****/settheclock/fastlane/Appfile'
- app_identifier: '*******'
- apple_id: '******'
- itc_team_id: '******'
- team_id: '******'
-------
INFO [2020-11-10 16:52:59.67]: Login to App Store Connect (***@***.**)
Reading keychain entry, because either user or password were empty
Loading session from '/Users/***/.fastlane/spaceship/****@home.nl/cookie'
Looking for App Store Connect Team with ID ****
INFO [2020-11-10 16:53:01.81]: Login successful

+--------------------------------------+------------------------+
|                    deliver 2.166.0 Summary                    |
+--------------------------------------+------------------------+
| skip_metadata                        | true                   |
| skip_binary_upload                   | true                   |
| skip_app_version_update              | true                   |
| overwrite_screenshots                | true                   |
| force                                | true                   |
| screenshots_path                     | ./fastlane/screenshots |
| metadata_path                        | ./fastlane/metadata    |
| app_version                          | 4.10                   |
| username                             | ***@****.**     |
| app_identifier                       | *****         |
| ipa                                  | ******.ipa      |
| platform                             | ios                    |
| edit_live                            | false                  |
| use_live_version                     | false                  |
| skip_screenshots                     | false                  |
| submit_for_review                    | false                  |
| reject_if_possible                   | false                  |
| phased_release                       | false                  |
| reset_ratings                        | false                  |
| team_id                              | *****                 |
| dev_portal_team_id                   | *****             |
| run_precheck_before_submit           | true                   |
| precheck_default_rule_level          | warn                   |
| ignore_language_directory_validation | false                  |
| precheck_include_in_app_purchases    | true                   |
+--------------------------------------+------------------------+

Traceback (most recent call last):
	24: from /usr/local/bin/fastlane:23:in `<main>'
	23: from /usr/local/bin/fastlane:23:in `load'
	22: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/bin/fastlane:23:in `<top (required)>'
	21: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane/lib/fastlane/cli_tools_distributor.rb:111:in `take_off'
	20: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:17:in `start'
	19: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:186:in `run'
	18: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	17: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	16: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	15: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	14: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	13: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:81:in `block (2 levels) in run'
	12: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/runner.rb:54:in `run'
	11: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/runner.rb:135:in `upload_metadata'
	10: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:254:in `collect_screenshots'
	 9: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:265:in `collect_screenshots_for_languages'
	 8: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:265:in `each'
	 7: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:289:in `block in collect_screenshots_for_languages'
	 6: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:289:in `each'
	 5: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:298:in `block (2 levels) in collect_screenshots_for_languages'
	 4: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:298:in `new'
	 3: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/app_screenshot.rb:89:in `initialize'
	 2: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/app_screenshot.rb:349:in `calculate_screen_size'
	 1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': Unsupported screen size [1284, 2778] for path './fastlane/screenshots/de-DE/iPhone 12 Pro Max-01-Game1.png' (FastlaneCore::Interface::FastlaneError)
	24: from /usr/local/bin/fastlane:23:in `<main>'
	23: from /usr/local/bin/fastlane:23:in `load'
	22: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/bin/fastlane:23:in `<top (required)>'
	21: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane/lib/fastlane/cli_tools_distributor.rb:111:in `take_off'
	20: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:17:in `start'
	19: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:186:in `run'
	18: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	17: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	16: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	15: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	14: from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	13: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/commands_generator.rb:81:in `block (2 levels) in run'
	12: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/runner.rb:54:in `run'
	11: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/runner.rb:135:in `upload_metadata'
	10: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:254:in `collect_screenshots'
	 9: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:265:in `collect_screenshots_for_languages'
	 8: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:265:in `each'
	 7: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:289:in `block in collect_screenshots_for_languages'
	 6: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:289:in `each'
	 5: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:298:in `block (2 levels) in collect_screenshots_for_languages'
	 4: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/upload_screenshots.rb:298:in `new'
	 3: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/app_screenshot.rb:89:in `initialize'
	 2: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/deliver/lib/deliver/app_screenshot.rb:349:in `calculate_screen_size'
	 1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': [!] Unsupported screen size [1284, 2778] for path './fastlane/screenshots/de-DE/iPhone 12 Pro Max-01-Game1.png' (FastlaneCore::Interface::FastlaneError)
</pre>
</details>

**Comment by ainame** [comment]
👋 

I've tested uploading screenshots locally. It seems that App Store Connect still accepts the old screenshot size  "1242 x 2688" (at least with web session). We would need to keep both old and new sizes supported. So the change shouldn't be just updating existing value but adding the new one with a different constant name?

https://github.com/fastlane/fastlane/blob/7e170280edb64b9941d5c4acd9eb41578e9e69bd/deliver/lib/deliver/app_screenshot.rb#L244-L247

**Comment by MrBenFTW** [comment]
I am also getting this error.

**Comment by ainame** [comment]
I submitted the PR for this🤞 
https://github.com/fastlane/fastlane/pull/17594

**Comment by rogerluan** [comment]
https://github.com/fastlane/fastlane/pull/17594 was merged and released as part of [_fastlane_ 2.167.0](https://github.com/fastlane/fastlane/releases/tag/2.167.0) 🚀 

Closing this issue 😊
